### PR TITLE
Adapted to class structure style policy - Part 6

### PR DESCRIPTION
**Pull request by thomasfanell**
This PR covers:
* source/sdk
* source/ui
* source/unit

### CODE REVIEW

File: source/sdk/Text.ooc [hunk 1]
```diff
@@ -291,6 +276,22 @@ Text: cover {
...
+	operator + (other: UInt) -> This { this + other toText() }
+	operator + (other: Float) -> This { this + other toText() }
+	operator + (other: Double) -> This { this + other toText() }
+	operator [] (index: Int) -> Char {
+		result := this _buffer[index]
+		this free(Owner Receiver)
+		result
+	}
+	operator [] (range: Range) -> This { this slice(range min, range count) }
+	empty: static This { get { This new() } }
```
Comment (marcusnaslund) [inline source/sdk/Text.ooc]: Can you do `::=` with `static`?

File: source/sdk/math.ooc [hunk 2]
```diff
@@ -191,18 +191,23 @@ extend Float {
...
 		result := this
 		if (result < 0)
 			result += ((result abs()) / divisor) ceil() * divisor
 		result mod(divisor)
 	}
+	equals: func (other: This, tolerance := This defaultTolerance) -> Bool { (this - other) abs() < tolerance }
+	lessThan: func (other: This, tolerance := This defaultTolerance) -> Bool { this < other && !this equals(other) }
+	greaterThan: func (other: This, tolerance := This defaultTolerance) -> Bool { this > other && !this equals(other) }
+	lessOrEqual: func (other: This, tolerance := This defaultTolerance) -> Bool { !this greaterThan(other) }
+	greaterOrEqual: func (other: This, tolerance := This defaultTolerance) -> Bool { !this lessThan(other) }
```
Comment (marcusnaslund) [inline source/sdk/math.ooc]: Maybe we can move these to the `LLong` cover in `Numbers.ooc`, and then we don't need to include them in Float, Double and LDouble.

File: source/sdk/threading/WaitCondition.ooc [hunk 3]
```diff
@@ -2,6 +2,11 @@ import native/[ConditionUnix, ConditionWin32]
 import Mutex
 
 WaitCondition: abstract class {
+	wait: abstract func (mutex: Mutex) -> Bool
+
+	signal: abstract func -> Bool
+
+	broadcast: abstract func -> Bool
```
Comment (marcusnaslund) [inline source/sdk/threading/WaitCondition.ooc]: Either include an empty line here or remove the two above

File: source/sdk/Text.ooc (hunk 1 above)
Comment (thomasfanell) [inline source/sdk/Text.ooc]: Yes, I've already noted this one down.

File: source/sdk/threading/ThreadLocal.ooc [hunk 4]
```diff
@@ -1,6 +1,11 @@
 import native/[ThreadLocalUnix, ThreadLocalWin32]
 
 ThreadLocal: abstract class <T> {
+	set: abstract func (value: T)
+
+	get: abstract func -> T
+
+	hasValue?: abstract func -> Bool
```
Comment (marcusnaslund) [inline source/sdk/threading/ThreadLocal.ooc]: Empty line here as well or remove the two above

File: source/sdk/threading/WaitCondition.ooc (hunk 3 above)
Comment (thomasfanell) [inline source/sdk/threading/WaitCondition.ooc]: Done.

File: source/sdk/threading/ThreadLocal.ooc (hunk 4 above)
Comment (thomasfanell) [inline source/sdk/threading/ThreadLocal.ooc]: Done (although most of these will be fixed in upcoming PRs)

File: source/unit/Fixture.ooc [hunk 5]
```diff
@@ -126,8 +117,19 @@ Fixture: abstract class {
 			result append(Text new(" [tolerance: %.8f]" formatDouble((constraint parent as CompareWithinConstraint) precision)))
 		result join(' ')
 	}
-	is ::= static IsConstraints new()
+	_testsFailed: static Bool
 	_expectCount: static Int = 0
+	is ::= static IsConstraints new()
+	testsFailed: static Bool { get { This _testsFailed } }
```
Comment (marcusnaslund) [inline source/unit/Fixture.ooc]: This one should also use `::=`

**Pull request merged**